torch/_inductor/codegen/cpp_wrapper_gpu.py

```diff
@@ -208,6 +208,27 @@ def write_get_raw_stream(self, device_idx: int, graph=None) -> str:
         )
         return name
 
+    def codegen_inputs(self):
+        # See Note: [Input Alignment handling in Inductor]
+        #
+        # JIT Inductor does not guard on input alignment. It relies on copy_misaligned_inputs to
+        # copy misaligned inputs to aligned buffers. For AOTInductor, we expect users to use it
+        # as non-Python deployment for its best performance, so implicitly copying misaligned inputs
+        # to aligned buffers is going to bring a surprising performance hit. Instead, we check input
+        # alignment and throw an error if any input is misaligned.
+        if V.graph.aot_mode:
+            for name, value in V.graph.graph_inputs.items():
+                if isinstance(value, TensorBox):
+                    self.prefix.splice(
+                        f"""
+                    if ((long({name}.data_ptr()) & ({GPU_ALIGN_BYTES} -1)) != 0) {{
+                        throw std::runtime_error("{name} is not aligned to {GPU_ALIGN_BYTES} bytes");
+                    }}
```

Comment:
Within inductor, we don't necessarily specialize on all inputs being aligned. We'll infer alignment from first invocation. It's taken from `inputs_to_check`.